src/PsrProducer.php

```diff
@@ -13,4 +13,55 @@
      * @throws InvalidMessageException     if an invalid message is specified
      */
     public function send(PsrDestination $destination, PsrMessage $message);
+
+    /**
+     * Sets the minimum length of time in milliseconds that must elapse after a message is sent before the provider may deliver the message to a consumer.
+     * deliveryDelay is set to zero by default.
+     *
+     * @param float|int $deliveryDelay - the delivery delay in milliseconds.
+     */
+    public function setDeliveryDelay($deliveryDelay);
```

Comment:
if the transport does not support it will throw an exception. 

JMS does not tell anything on what transport should do in this case (when it does not support a feature)